nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/OAuth2Object.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * (C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Gabriel Barata <gbarata@nuxeo.com>
+ */
+package org.nuxeo.ecm.restapi.server.jaxrs;
+
+import com.google.api.client.auth.oauth2.Credential;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.Response.StatusType;
+
+import org.nuxeo.ecm.automation.server.jaxrs.RestOperationException;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.platform.oauth2.providers.AbstractOAuth2UserEmailProvider;
+import org.nuxeo.ecm.platform.oauth2.providers.NuxeoOAuth2ServiceProvider;
+import org.nuxeo.ecm.platform.oauth2.providers.OAuth2ServiceProvider;
+import org.nuxeo.ecm.platform.oauth2.providers.OAuth2ServiceProviderRegistry;
+import org.nuxeo.ecm.platform.oauth2.tokens.NuxeoOAuth2Token;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.AbstractResource;
+import org.nuxeo.ecm.webengine.model.impl.ResourceTypeImpl;
+import org.nuxeo.runtime.api.Framework;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Endpoint to retrieve OAuth2 authentication data
+ * @since 8.4
+ */
+@WebObject(type = "oauth2")
+public class OAuth2Object extends AbstractResource<ResourceTypeImpl> {
+
+    /**
+     * Retrieves oauth2 data for a given provider.
+     */
+    @GET
+    @Path("provider/{providerId}")
+    public Response getProvider(@PathParam("providerId") String providerId,
+                                @Context HttpServletRequest request) throws IOException, RestOperationException {
+
+        NuxeoOAuth2ServiceProvider provider = getProvider(providerId);
+
+        Map<String,Object> result = new HashMap<>();
+        result.put("serviceName", provider.getServiceName());
+        result.put("enabled", provider.isEnabled());
```

Comment:
Would isProviderAvailable() be more relevant ? It checks for clientId and clientSecret as well